### ISSUE TITLE
Struggling with network training

### ISSUE THREAD

**Issue by xburda4** [cbovar/ConvNetSharp#73]
Hello,

I am trying to train network such that it will find some object in image. For input I created Volume(someArray,Shape(64,64,1,batchSize)). Images are grayscale with size 64x64. For someArray I created array of size 64*64*batchSize and it is filled with values of pixels such that every new image begins at position k*4096.

As output layer I used Softmax. Idea was that for every column in image I create separate class. For some reason though, during training I get "Error during calculation". This is because network will eventually adjust its weights, so that all weights coming from convolutional layer are set to very big numbers and then to Infinity. 

So I wanted to ask, am I working with volumes right? If I want to train network on multiple images, do I do it in a way, that all images are in one array,one image after another, and network will know how to parse the array according to volume's dimensions? Can someone point me where could I look for errors in my code?

**Comment by cbovar** [comment]
Hi,

- You could try to decrease the learning rate. This may prevent the gradient explosion
- I think you are using volumes the correct way. [64x64x1xBatchSize] is the correct shape. If you want to teach using more sample just present another volume to the network (you don't have to fit all your data into one batch).

**Comment by xburda4** [comment]
Thank you for answering. So I set learning rate to 0.001 and it looks like it might work. I also had to modify input values so they would be between 0 and 1.

In case I would want to teach network using more batches, I could fit it into one cycle with multiple net.Train() commands or I would just had more cycles where I would train one batch, than the others?

**Comment by cbovar** [comment]
You could look at Mnist demo to get some inspiration:
* [The learning loop](https://github.com/cbovar/ConvNetSharp/blob/master/Examples/MnistDemo/Program.cs#L55)
* [Batch construction](https://github.com/cbovar/ConvNetSharp/blob/master/Examples/MnistDemo/DataSet.cs#L20)

It is one cycle that trains one batch at the time.
